### PR TITLE
PushoverAgent: HTML message support

### PR DESCRIPTION
**Pull request by dgw**
Pushover has supported displaying the `message` as HTML in its apps for about three years. https://updates.pushover.net/post/117525190347/html-message-support

### CODE REVIEW

File: app/models/agents/pushover_agent.rb [hunk 1]
```diff
@@ -30,6 +30,7 @@ class PushoverAgent < Agent
       * `sound` - the name of one of the sounds supported by device clients to override the user's default sound choice. [See PushOver docs for sound options.](https://pushover.net/api#sounds)
       * `retry` - Required for emergency priority - Specifies how often (in seconds) the Pushover servers will send the same notification to the user. Minimum value: `30`
       * `expire` - Required for emergency priority - Specifies how many seconds your notification will continue to be retried for (every retry seconds). Maximum value: `86400`
+      * `html` - set to `1` to have Pushover's apps display the `message` content as HTML
```
Comment (knu) [inline app/models/agents/pushover_agent.rb]: Maybe this should be a `true`/`false` field, following our conventions.

File: app/models/agents/pushover_agent.rb [hunk 2]
```diff
@@ -84,6 +86,7 @@ def receive(incoming_events)
             sound
             retry
             expire
+            html
           ].each do |key|
             if value = String.try_convert(interpolated[key].presence)
               case key
```
Comment (knu) [inline app/models/agents/pushover_agent.rb]: Add the following lines here to accept `true` as a value for `html`:
```ruby
when 'html'
  case value
  when 'true', '1'
    value = '1'
  else
    next
  end
```

Comment (dgw) [reply app/models/agents/pushover_agent.rb]: I did mostly this, but explicitly set `value = '0'` in the `else` instead of just going to the `next` key.

**Comment by dgw** [comment]
@knu Addressed your comments, let me know if you want the two commits squashed into one. :)

**Comment by knu** [comment]
@dgw Great!  And yes, please squash them.

**Comment by knu** [comment]
Oh wait, if `html` were set to `true` instead of `"true"` it would not count.  You'd need to treat `html` separately.  For example, outside of the each loop:

```ruby
if value = interpolated['html'].presence
  post_params['html'] =
    case value.to_s
    when 'true', '1'
      '1'
    else
      '0'
    end
end
```

**Comment by knu** [comment]
(That is because `String.try_convert(true) == nil`.)

**Comment by knu** [comment]
There’s nothing wrong with try_convert, it is meant to work that way.  Anyway, thanks for your contribution, I’ll try out HTML messages!

**Comment by dgw** [comment]
Ha, I was going to squash at least the first two commits on approval. Oh well.

I don't understand why `String.try_convert(true)` is meant to return `nil` for a value that should be easily converted to String, but I'm new to Ruby. Haven't yet gotten my head around the difference between `to_s` and `to_str`, the distinction between explicit and implicit conversion, and how `try_convert` works.

File: app/models/agents/pushover_agent.rb [hunk 3]
```diff
@@ -95,6 +97,16 @@ def receive(incoming_events)
               post_params[key] = value
             end
           end
+          # html is special because String.try_convert(true) gives nil (not even "nil", just nil)
+          if value = interpolated['html'].presence
+            post_params['html'] =
+              case value.to_s
+              when 'true', '1'
```
Comment (dsander) [inline app/models/agents/pushover_agent.rb]: Is there a reason why we want to support `1` as true here? We normally rely on the [`boolify`](https://github.com/huginn/huginn/blob/fdf4b0794aaed8a342d0dbc1658b43f52814982b/app/models/agent.rb#L281) that only accepts `true` and `"true"`.

Comment (dgw) [reply app/models/agents/pushover_agent.rb]: Pushover's API documentation gives `1` as the value to pass, so it made sense to support what their docs said to do.

**Pull request merged**